### DISCUSSION TITLE
blocking incosistent

### DISCUSSION THREAD

**Discussion by ghajini** [uBlockOrigin/uBlock-issues#2764] [Q&A]
@gorhill 
i do have inconsistent blocking by ublock origin
when i force update filters ,open website ,ublock doesn't block anything with respect to filters...
when i force reload, everything works then , again when i force update filter lists ,it again shows no blocking

video is attached

```
uBlock Origin: 1.51.1b7
Chromium: 115
filterset (summary):
 network: 99424
 cosmetic: 41670
 scriptlet: 22736
 html: 0
listset (total-discarded, last-updated):
 removed:
  easylist: null
 added:
  https://easylist.to/easylist/easylist.txt: 69392-134, 14m
  ublock-annoyances: 5323-14, 14m
 default:
  user-filters: 2894-11, never
  easyprivacy: 33320-65, 14m
  plowe-0: 3716-1069, 14m
  ublock-badware: 7679-143, 14m
  ublock-filters: 34155-611, 14m
  ublock-privacy: 447-4, 14m
  ublock-quick-fixes: 274-42, 14m
  ublock-unbreak: 2106-37, 14m
  urlhaus-1: 6826-0, 14m
filterset (user): [array of 2032 redacted]
trustedset:
 added: [array of 4 redacted]
 removed:
  about-scheme
  chrome-extension-scheme
  chrome-scheme
  edge-scheme
switchRuleset:
 added: [array of 1 redacted]
userSettings:
 advancedUserEnabled: true
 suspendUntilListsAreLoaded: true
 webrtcIPAddressHidden: true
hiddenSettings:
 filterAuthorMode: true
 updateAssetBypassBrowserCache: true
 userResourcesLocation: [redacted]
supportStats:
 allReadyAfter: 2079 ms (selfie)
 maxAssetCacheWait: 429 ms
```

https://github.com/uBlockOrigin/uBlock-issues/assets/20338483/a8432343-a709-4bb4-b7ec-526b80cceb91

**Comment by gorhill** [comment]
When filter lists are updated, uBO must reload all filter lists in memory, and there is a time window within which there won't be filters, and if uBO looks up filters, nothing will be returned. The size of the time window depends on your computer performance. On my side I typically get under 200 ms on my side to reload all filter lists in memory. I suspect you loaded the page while inside that filter lists-reload time window in uBO.

**Reply by ghajini** [reply]
ok on my side , i had to wait 22 seconds to reload all filter lists after reading filter lists(force update)
what's the fix for this?

**Reply by gorhill** [reply]
The critical time window is after the filter lists all updated, not during. If filter lists take long to update, it's up to your network connection.  Do your test again, but once lists are all updated, wait a few seconds before reloading the page. If this doesn't work, repeat but wait longer. Eventually I expect there will be a waiting delay for which there is no filtering issue. That would confirm my hypothesis.

**Reply by u-RraaLL** [reply]
Shouldn't the `allReadyAfter` value change after purging and updating filter lists? And if it did update, wouldn't that re-trigger the icon/popup readiness check which would than warn users if uBO wasn't ready directly after an update?
